Sources/MongoSwift/Operations/ListCollectionsOperation.swift

```diff
@@ -117,38 +117,36 @@ internal struct ListCollectionsOperation: Operation {
             }
         }
 
-        let indexes: OpaquePointer = self.database.withMongocDatabase(from: connection) { dbPtr in
+        let collections: OpaquePointer = self.database.withMongocDatabase(from: connection) { dbPtr in
             guard let collections = mongoc_database_find_collections_with_opts(dbPtr, opts._bson) else {
                 fatalError(failedToRetrieveCursorMessage)
             }
             return collections
         }
 
         if self.nameOnly {
-            let cursor: MongoCursor<Document> = try MongoCursor(
-                stealing: indexes,
+            let cursor = try Cursor(
+                mongocCursor: MongocCursor(referencing: collections),
                 connection: connection,
-                client: self.database._client,
-                decoder: self.database.decoder,
-                session: session
+                session: session,
+                type: .nonTailable
             )
-            defer { cursor.blockingKill() }
+            defer { cursor.kill() }
 
             var names = [String]()
             // call the cursor method for getting the next document directly rather than just iterating through the
```

Comment:
this comment seems a little out of date since there's no `NextOperation` anymore, maybe update it to explain what you mentioned in your comment above about using the internal cursor type